### PR TITLE
matomo: 4.5.0 -> 4.7.1, module improvements

### PR DESCRIPTION
**Pull request by abbradar**
###### Motivation for this change
Update Matomo and add more automatic setup, like MariaDB database.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Tested by running the service and checking that both setup and system check pass more tests.

Depends on https://github.com/NixOS/nixpkgs/pull/100617.

### CODE REVIEW

File: nixos/modules/services/web-apps/matomo.nix [hunk 1]
```diff
@@ -113,13 +113,22 @@ in {
...
+          Either this option or the webServerUser option is mandatory.
+          Set this to {} to just enable the virtualHost if you don't need any customization.
+          If enabled, then by default, the <option>serverName</option> is
+          <literal>''${user}.''${config.networking.hostName}.''${config.networking.domain}</literal>,
+          SSL is active, and certificates are acquired via ACME.
+          If this is set to null (the default), no nginx virtualHost will be configured.
+        '';
+      };
+
+      mariaDB = mkOption {
```
Comment (aanderse) [inline nixos/modules/services/web-apps/matomo.nix:126]: Often times we wrap this behind a few options like `database.createLocally`, `database.name`, etc...

... if you're interested.

Comment (abbradar) [reply nixos/modules/services/web-apps/matomo.nix]: Making this an attrset from the beginning is a good idea for backcomp! Did that, added `name` option. Sadly there's no good way to configure this declaratively for Matomo (we may implement this in the future, but it's difficult), but one can at least change created local database name.

**Comment by sebbel** [comment]
hey @abbradar, love the magic option for the database :)

**Comment by abbradar** [comment]
Resolved conflicts and added a patch for https://github.com/matomo-org/matomo/issues/18785, dropping previous workaround. Let's merge this in a few days if noone objects.

**Review by sebbel — APPROVED** [review]

**Comment by bbenno** [comment]
Any news on updating Matomo to v4.7.1?

File: pkgs/servers/web-apps/matomo/default.nix [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+      sha256 = "sha256-MuuCKWKIzf4V7JmV4y2EH8oONSetJyciKmLNpHdC1g8=";
     };
 
-    matomo-beta = {
+    matomo-beta = matomo; /* {
       version = "4.6.0";
       # `beta` examples: "b1", "rc1", null
       # when updating: use null if stable version is >= latest beta or release candidate
       beta = "b2";
       sha256 = "sha256-7p/ZPtr5a/tBjrM27ILF3rNfxDIWuzWKCXNom3HlyL8=";
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/matomo/default.nix:15]:
```suggestion
      sha256 = "";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
please rebase

**Comment by Janik-Haag** [comment]
@abbradar can you please re base this like Sandro already asked you to. I can give it a try if you want me to or close the pr If you don't find time to do so or don't respond in the few weeks.

**Pull request closed without merging**